### ISSUE TITLE
twisted/test/test_stringtransport.py fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8719]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8719|
|Type|enhancement|
|Created|2016-08-02 11:32:26Z|
|Branch|https://github.com/twisted/twisted/tree/8719-rodrigc-stringtransport-py3|

If I try to run:

tox -e py35-tests twisted/test/test_stringtransport.py

it fails.

<details><summary>Searchable metadata</summary>

```
trac-id__8719 8719
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8719_rodrigc_stringtransport_py3 8719-rodrigc-stringtransport-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1470137546944516 1470137546944516
changetime__1470651271493472 1470651271493472
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/469

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_2"></a>|@acabhishek942 commented|
|-|-|

changes looks good. Please merge

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 6fafccd6ca7f8821f1a25bb4aa1801f4802833f8
```
#!CommitTicketReference repository="" revision="6fafccd6ca7f8821f1a25bb4aa1801f4802833f8"
Merge pull request #469 from twisted/8719-rodrigc-stringtransport-py3

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8719

Fix twisted/test/test_stringtransport.py  on Python 3
```
